### PR TITLE
perf(@angular-devkit/build-angular): disable `showCircularDependencies` by default

### PR DESCRIPTION
**Pull request by alan-agius4**
Change the default of this flag to disabled by default as this is a caused of rebuild performance degradation for large projects.

BREAKING CHANGE

The server and browser builder `showCircularDependencies` option default value has been changed from `true` to `false`.

### CODE REVIEW

File: packages/angular_devkit/build_angular/src/browser/schema.json [hunk 1]
```diff
@@ -290,7 +290,8 @@
     "showCircularDependencies": {
       "type": "boolean",
       "description": "Show circular dependency warnings on builds.",
-      "default": true
+      "default": false,
+      "x-deprecated": "Use a linting rule instead instead."
```
Comment (JoostK) [inline packages/angular_devkit/build_angular/src/browser/schema.json:294]:
```suggestion
      "x-deprecated": "Use a linting rule instead."
```

File: packages/angular_devkit/build_angular/src/server/schema.json [hunk 2]
```diff
@@ -187,7 +187,8 @@
     "showCircularDependencies": {
       "type": "boolean",
       "description": "Show circular dependency warnings on builds.",
-      "default": true
+      "default": false,
+      "x-deprecated": "Use a linting rule instead instead."
```
Comment (JoostK) [inline packages/angular_devkit/build_angular/src/server/schema.json:191]:
```suggestion
      "x-deprecated": "Use a linting rule instead."
```

Comment (alan-agius4) [reply packages/angular_devkit/build_angular/src/server/schema.json]: Was still sleeping :)

File: packages/angular_devkit/build_angular/src/browser/schema.json [hunk 3]
```diff
@@ -290,7 +290,8 @@
     "showCircularDependencies": {
       "type": "boolean",
       "description": "Show circular dependency warnings on builds.",
-      "default": true
+      "default": false,
+      "x-deprecated": "Use a linting rule instead."
```
Comment (clydin) [inline packages/angular_devkit/build_angular/src/browser/schema.json:294]:
```suggestion
      "x-deprecated": "The recommended method to detect circular dependencies in project code is to use a either a lint rule or other external tooling."
```

File: packages/angular_devkit/build_angular/src/browser/tests/options/show-circular-dependencies_spec.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  describe('Option: "showCircularDependencies"', () => {
+    beforeEach(async () => {
+      // Add circular dependency
+      await harness.appendToFile(
+        'src/app/app.component.ts',
+        `import { AppModule } from './app.module'; console.log(AppModule);`,
+      );
+    });
+
+    it('should show cylic dependency warning when option is set to true', async () => {
```
Comment (clydin) [inline packages/angular_devkit/build_angular/src/browser/tests/options/show-circular-dependencies_spec.ts:22]:
```suggestion
    it('should show cyclic dependency warning when option is set to true', async () => {
```

**Review by clydin — APPROVED** [review]

**Pull request merged**